docs: add FAQ for linting errors

ShabiShett07 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: na
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: na
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes. report:
  - task: run_javascript_examples status: na
  - task: run_c_examples status: na
  - task: run_cpp_examples status: na
  - task: run_javascript_readme_examples status: na
  - task: run_c_benchmarks status: na
  - task: run_cpp_benchmarks status: na
  - task: run_fortran_benchmarks status: na
  - task: run_javascript_benchmarks status: na
  - task: run_julia_benchmarks status: na
  - task: run_python_benchmarks status: na
  - task: run_r_benchmarks status: na
  - task: run_javascript_tests status: na ---

none
## Description

> What is the purpose of this pull request?

This pull request:

-   {Adds a FAQ for contributing FAQs document

## Related Issues

> Does this pull request have any related issues?

This pull request:

-  none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
docs: add faq

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

ShabiShett07 commented:
@aayush0325, please check this FAQ.

Also I think we should have a FAQ for max length in javascript files too. What is your opinion?

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -28,6 +28,7 @@ limitations under the License.
 -   [How can I install cppcheck?](#install-cppcheck)
 -   [I am seeing different return values in the JavaScript and C implementation for the same implementation.](#js-vs-c-return-values)
 -   [What should I do if Markdown linting on my commits fails because my headings exceed the maximum permissible length?](#markdown-heading-length)
+-   [What should I do if Javascript linting on my commits fails because my function exceed the maximum permissible number of parameters?](#max-params)
```
```suggestion
-   [What should I do if linting on my commits fails due to Markdown headings exceeding maximum length or JavaScript functions exceeding maximum parameters?](#linting-errors)
```

@ShabiShett07 I think that's a nice addition:

We can probably combine these two into one and then include the respective commands for JS/Markdown.

anandkaranubc replied:
Maybe or maybe not. Open to feedback.

ShabiShett07 replied:
That's a good idea, but max params and max length are two different conditions, right? I think we can include max length of javascript in the markdown FAQ. What is your opinion @anandkaranubc?

anandkaranubc replied:
> I think we can include max length of javascript...

Even better, I believe.  

> but max params and max length are two different conditions

And you are right, JS/Markdown linting questions are different here. I realized this soon after adding the above comment.

EDIT: Can you also add the reference PRs that contain these comments?

aayush0325 replied:
my take on this is:

1. max-len - one heading for this and we should describe how to tackle this in various places, for eg we would use `// eslint-disable-line max-len` in JS, `<--! lint-disable max-len --->` or something in markdown (not sure if the command is correct here please cross check). and for places like `repl.txt` we can just add tips on how to get past the 80 char limit, by shortening the variable names, removing the spaces after each param if needed, line breaks after 80 chars etc
2. max-params - similar to `max-len`

TLDR - one FAQ each for max-len and max-params which describe how to deal with these warning in all places like JS/TS, markdown, repl.txt etc

ShabiShett07 replied:
I think the same too.

ShabiShett07 replied:
For reference PR can I add my PR to it

anandkaranubc replied:
> TLDR - one FAQ each for max-len and max-params which describe how to deal with these warning in all places like JS/TS, markdown, repl.txt etc

+1. Thanks @aayush0325

aayush0325 replied:
i've used all of these extensively in #5754, you can use that as a reference

ShabiShett07 replied:
Sure

ShabiShett07 committed:
chore: add reference and faq

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
@aayush0325, I have added reference links as well as a FAQ for max length of javascript. Also I have removed a TODO in line 68 because I thought it was outdated. Please check if I am right.

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
 <a name="markdown-heading-length"></a>
 
-## What should I do if Markdown linting on my commits fails because my headings exceed the maximum permissible length?
+## What should I do if linting on my commits fails because my headings exceed the maximum permissible length?
```
```suggestion
## What should I do if linting on my commits fails because my headings or lines exceed the maximum permissible length?
```

Maybe, we can be more explicit...

ShabiShett07 replied:
Added this in the FAQ

anandkaranubc commented on docs/contributing/FAQ.md:
````diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
+// eslint-disable-line max-params
+```
+
+- [Reference PR for Javascript][javascript-params-ref]
````
```suggestion
[Reference PR][javascript-params-ref]
```

Can you link to the exact line here?

anandkaranubc commented on docs/contributing/FAQ.md:
````diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
+// eslint-disable-line max-len
+```
+
+- [Reference PR for Javascript][javascript-len-ref]
````
```suggestion
[Reference PR][javascript-len-ref]
```

And here too :)

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -28,6 +28,7 @@ limitations under the License.
 -   [How can I install cppcheck?](#install-cppcheck)
 -   [I am seeing different return values in the JavaScript and C implementation for the same implementation.](#js-vs-c-return-values)
 -   [What should I do if Markdown linting on my commits fails because my headings exceed the maximum permissible length?](#markdown-heading-length)
```
This will also be changed accordingly.

ShabiShett07 committed:
fix: unmatched errors

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: change mismatched words

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -386,6 +407,12 @@ For more `make` commands, refer to the [documentation][benchmark] on running ben
 
 [make-commands]: https://github.com/stdlib-js/stdlib/tree/develop/tools/make/lib
 
+[markdown-len-ref]: https://github.com/stdlib-js/stdlib/pull/5754/files#diff-8b632afc40c671b4097b2d9fc318b0b6eedcebddeff0e13aad0a4d68260505d8R94
```
```suggestion
[markdown-len-ref]: https://github.com/stdlib-js/stdlib/blob/78e0cfd8b6c0429a443b07fd39fa9dd53bf44d23/lib/node_modules/%40stdlib/lapack/base/dgttrf/README.md?plain=1#L94
```

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -386,6 +407,12 @@ For more `make` commands, refer to the [documentation][benchmark] on running ben
 
+[markdown-len-ref]: https://github.com/stdlib-js/stdlib/pull/5754/files#diff-8b632afc40c671b4097b2d9fc318b0b6eedcebddeff0e13aad0a4d68260505d8R94
+
+[javascript-len-ref]: https://github.com/stdlib-js/stdlib/pull/5754/files#diff-71a4cffefbc20405eaf647ea47353b5561fd10fc032a214183f0c307c3dc4624R75
```
```suggestion
[javascript-len-ref]: https://github.com/stdlib-js/stdlib/blob/78e0cfd8b6c0429a443b07fd39fa9dd53bf44d23/lib/node_modules/%40stdlib/lapack/base/dgttrf/lib/base.js#L111
```

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -386,6 +407,12 @@ For more `make` commands, refer to the [documentation][benchmark] on running ben
+
+[javascript-len-ref]: https://github.com/stdlib-js/stdlib/pull/5754/files#diff-71a4cffefbc20405eaf647ea47353b5561fd10fc032a214183f0c307c3dc4624R75
+
+[javascript-params-ref]: https://github.com/stdlib-js/stdlib/pull/5754/files#diff-71a4cffefbc20405eaf647ea47353b5561fd10fc032a214183f0c307c3dc4624R75
```
```suggestion
[javascript-params-ref]: https://github.com/stdlib-js/stdlib/blob/78e0cfd8b6c0429a443b07fd39fa9dd53bf44d23/lib/node_modules/%40stdlib/lapack/base/dgttrf/lib/base.js#L75
```

anandkaranubc replied:
@ShabiShett07 @aayush0325  

These links will need to be revisited once Aayush's PR is merged so that we can reference the respective lines in the `stdlib` repo itself.

aayush0325 replied:
everything looks good to me, we can just add a todo and come back to this

anandkaranubc commented on docs/contributing/FAQ.md:
````diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
+// eslint-disable-line max-len
+```
+
+[Reference PR][javascript-len-ref]
````
```suggestion
[Reference Comment][javascript-len-ref]
```

anandkaranubc commented on docs/contributing/FAQ.md:
````diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
 ```
 
-TODO: Can we add a reference PR link?
+[Reference PR][markdown-len-ref]
````
```suggestion
[Reference Comment][markdown-len-ref]
```

ShabiShett07 committed:
chore: add correct link

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: appropriate word

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: change len ref link

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: change params ref link

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: change of word

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

anandkaranubc commented on docs/contributing/FAQ.md:
````diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
+// eslint-disable-line max-params
+```
+
+[Reference PR][javascript-params-ref]
````
```suggestion
[Reference Comment][javascript-params-ref]
```

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
-## What should I do if Markdown linting on my commits fails because my headings exceed the maximum permissible length?
+## What should I do if linting on my commits fails because my headings or lines exceed the maximum permissible length?
 
 Consider whether the heading can be shortened by renaming variables (e.g., changing `strideX` to `sx`). If shortening is not possible, disable the lint rule at the top level using:
```
```suggestion
Consider whether the heading/line can be shortened by renaming variables (e.g., changing `strideX` to `sx`). If shortening is not possible, disable the lint rule at the top level using:
```

Sorry, I’m being too nitpicky.

ShabiShett07 replied:
Actually, thank you for such interest in this.

ShabiShett07 committed:
fix: word error

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: added appropriate format

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -27,7 +27,8 @@ limitations under the License.
 -   [I am seeing different return values in the JavaScript and C implementation for the same implementation.](#js-vs-c-return-values)
--   [What should I do if Markdown linting on my commits fails because my headings exceed the maximum permissible length?](#markdown-heading-length)
+-   [What should I do if linting on my commits fails because my headings or lines exceed the maximum permissible length?](#markdown-heading-length)
+-   [What should I do if Javascript linting on my commits fails because my function exceed the maximum permissible number of parameters?](#max-params)
```
```suggestion
-   [What should I do if JavaScript linting on my commits fails because my function exceeds the maximum permissible number of parameters?](#max-params)
```

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
+
+Consider whether the heading/line can be shortened by renaming variables (e.g., changing `strideX` to `sx`). If shortening is not possible, disable the lint rule at the top level using:
+
+- For Javascript Files
```
```suggestion
- For JavaScript files:
```

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
+
+[Reference Comment][javascript-len-ref]
+
+- For Markdown Files
```
```suggestion
- For Markdown files:
```

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
+
+<a name="markdown-heading-length"></a>
+
+## What should I do if Javascript linting on my commits fails because my headings exceed the maximum permissible number of parameters?
```
```suggestion
## What should I do if JavaScript linting on my commits fails because my function exceeds the maximum permissible number of parameters?
```

anandkaranubc replied:
@ShabiShett07 I also added these review comments.

anandkaranubc reviewed (changes requested):
Some last changes :) @ShabiShett07 

Let's also change the title to:

docs: add FAQ for linting errors

ShabiShett07 commented:
> Some last changes :) @ShabiShett07
> 
> Let's also change the title to:
> 
> docs: add FAQ for linting errors

I have changed the title

anandkaranubc reviewed (changes requested):
The above comments still need to be resolved first.

ShabiShett07 committed:
chore: add suggestions

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add suggestions

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add suggestions

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add suggestions

Co-authored-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
@anandkaranubc, I think I have solved all your comments, sorry for not noticing the comments

anandkaranubc commented on docs/contributing/FAQ.md:
```diff
@@ -101,15 +100,37 @@ If they pass, adjust the tolerance and add a note to the C tests indicating that
+
+## What should I do if JavaScript linting on my commits fails because my function exceeds the maximum permissible number of parameters?
+
+Consider whether the number of parameters can be reduced. If reducing is not possible, disable the lint rule at the top level using:
```
```suggestion
Consider whether the number of parameters can be reduced. If reduction is not possible, disable the lint rule at the top level using:
```

anandkaranubc committed:
chore: apply suggestions from review

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>
